bundles/org.openhab.binding.ojelectronics/src/main/java/org/openhab/binding/ojelectronics/internal/services/RefreshService.java

```diff
@@ -0,0 +1,158 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.ojelectronics.internal.services;
+
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.client.api.Result;
+import org.eclipse.jetty.client.util.BufferingResponseListener;
+import org.eclipse.jetty.http.HttpMethod;
+import org.eclipse.jetty.http.HttpStatus;
+import org.openhab.binding.ojelectronics.internal.config.OJElectronicsBridgeConfiguration;
+import org.openhab.binding.ojelectronics.internal.models.groups.GroupContentResponseModel;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonSyntaxException;
+
+/**
+ * Handles the refreshing of the devices of a OJElectronics session
+ *
+ * @author Christian Kittel - Initial Contribution
+ */
+@NonNullByDefault
+public final class RefreshService implements AutoCloseable {
+
+    private final OJElectronicsBridgeConfiguration config;
+    private final Logger logger = LoggerFactory.getLogger(RefreshService.class);
+    private final HttpClient httpClient;
+    private final Gson gson = new GsonBuilder().setPrettyPrinting().setDateFormat("yyyy-MM-dd'T'HH:mm:ss").create();
+    private final ScheduledExecutorService schedulerService;
+
+    private @Nullable Runnable connectionLost;
+    private @Nullable Consumer<@Nullable GroupContentResponseModel> refreshDone;
+    private @Nullable ScheduledFuture<?> scheduler;
+    private @Nullable Runnable unauthorized;
+    private String sessionId = "";
+    private static Boolean DESTROYED = false;
```

Comment:
Didn't notice this can't be final. In this case this should be lower case. Sorry.

You could use the primitve type: `boolean`